test/modules/encoders/test_clip_text_encoder.py

```diff
@@ -12,110 +12,100 @@
         assert isinstance(text_encoder, torch.nn.Module)
 
-        out = text_encoder(text)
-        assert out.shape == (batch_size, embedding_dim)
-        assert_expected(out.sum().item(), -2.3848, rtol=0, atol=1e-3)
-
-    def test_small_embedding(self, setUp):
-        context_length = setUp
-        batch_size, embedding_dim = 4, 1
-        text_length = min(context_length, 3)
-        text = torch.randint(1, 10, (batch_size, text_length), dtype=torch.long)
-
-        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
-        assert isinstance(text_encoder, torch.nn.Module)
+        actual_no_init = text_encoder(text)
+        expected_no_init = torch.Tensor(
+            [[-0.3195, 0.1647, -0.2621, 0.4387], [-0.3354, 0.2859, -0.6949, 0.2047]]
+        )
+        assert_expected(
+            actual=actual_no_init, expected=expected_no_init, rtol=0, atol=1e-4
+        )
 
-        out = text_encoder(text)
-        assert out.shape == (batch_size, embedding_dim)
-        assert_expected(out.sum().item(), -7.8039, rtol=0, atol=1e-3)
+        # Standard deviation of token embedding weights should be different without vs with CLIP params
+        actual_std_no_init = torch.std(
+            text_encoder.encoder.token_embedding.weight
+        ).item()
+        expected_std_clip_init = CLIPTextEncoder.TOKEN_EMBEDDING_INIT_STD
+        assert abs(actual_std_no_init - expected_std_clip_init) > 1e-4
 
-    def test_small_batch(self, setUp):
-        context_length = setUp
-        batch_size, embedding_dim = 1, 8
-        text_length = min(context_length, 3)
-        text = torch.randint(1, 10, (batch_size, text_length), dtype=torch.long)
+        # Then test the default CLIP initialization
+        text_encoder.initialize_parameters()
 
-        text_encoder = CLIPTextEncoder(embedding_dim=embedding_dim, use_clip_init=True)
-        assert isinstance(text_encoder, torch.nn.Module)
+        actual_clip_init = text_encoder(text)
+        expected_clip_init = torch.Tensor(
+            [[-1.9412, -0.8044, -0.7616, -0.3185], [0.2408, -0.3241, -0.2544, -0.3608]]
+        )
+        assert_expected(
+            actual=actual_clip_init, expected=expected_clip_init, rtol=0, atol=1e-4
+        )
 
-        out = text_encoder(text)
-        assert out.shape == (1, embedding_dim)
-        assert_expected(out.sum().item(), -1.4762, rtol=0, atol=1e-3)
+        actual_std_clip_init = torch.std(
+            text_encoder.encoder.token_embedding.weight
+        ).item()
+        assert_expected(
+            actual=actual_std_clip_init,
+            expected=expected_std_clip_init,
+            rtol=0,
+            atol=1e-4,
+        )
 
-    def test_at_context_length(self, setUp):
-        context_length = setUp
-        batch_size, embedding_dim = 4, 8
-        text = torch.randint(1, 10, (batch_size, context_length), dtype=torch.long)
+    def test_at_context_length(self, start):
```

Comment:
Is there any significant difference due to `text_length` and `context_length`? If the purpose is just to make sure the code runs when `text_length <= context_length`, shall we just have one test (e.g., `test_forward`) since you already have a test for the edge case `over_context_length`?